backend/routes/knowledge.py

```
@@ -0,0 +1,736 @@
+        }
+
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Failed to update file content: {e}", exc_info=True)
+        raise HTTPException(status_code=500, detail=f"Failed to update file content: {str(e)}")
+
+
+@router.post("/knowledge/upload")
+async def upload_knowledge_files(
+    request: Request, files: List[UploadFile] = File(...), session: dict = Depends(require_admin_auth)
```

Ruff (0.12.2)

Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)